### PR TITLE
fix(appconnect): Correct parsing builds form API

### PR DESCRIPTION
**Pull request by flub**
The last refactor here broke the fetching of build info because it was
accessing the build number from an attribute which did not exist.

Also re-write this so that these errors are easier to debug and errors
result in more fatal exceptions which are reported to the internal
sentry.  It still tries to process all the items in a single page however,
which means if there is an actual coding error we get an issue with
lots of events (depending on the number of builds).  On the other
hand if there are errors on processing the page-level elements then
the entire task fails and no attempt is made to process the next page.

@Swatinem next time please test locally 😉 
(yes, there is a desperate lack of tests for these things.  a debt we will have to pay at some point because otherwise things like this will keep breaking production)

### CODE REVIEW

File: src/sentry/utils/appleconnect/appstore_connect.py [hunk 1]
```diff
@@ -141,42 +138,56 @@ def get_build_info(
...
+        for build in page["data"]:
+            try:
+                related_appstore_version = get_related(build, "appStoreVersion")
+                related_prerelease_version = get_related(build, "preReleaseVersion")
+                related_version = related_appstore_version or related_prerelease_version
+                if not related_version:
+                    logger.error("Missing related version for AppStoreConnect `build`")
+                    continue
+                platform = related_version["attributes"]["platform"]
+                version = related_version["attributes"]["version"]
```
Comment (Swatinem) [inline src/sentry/utils/appleconnect/appstore_connect.py:182]: So this is the only real change here?
I did test this locally and had it working. Maybe I did some more refactoring before committing.
Is the field named like that for both `appStoreVersion` and `preReleaseVersion`?

**Review by Swatinem — APPROVED** [review]

File: src/sentry/utils/appleconnect/appstore_connect.py (hunk 1 above)
Comment (flub) [reply src/sentry/utils/appleconnect/appstore_connect.py:182]: 😱  that was an excellent question!

https://developer.apple.com/documentation/appstoreconnectapi/appstoreversion/attributes
https://developer.apple.com/documentation/appstoreconnectapi/prereleaseversion/attributes

And indeed the reason this error happened.  let's try this again.

**Comment by flub** [comment]
@Swatinem PTAL, i have only tested an app with preReleaseVersion builds, so the appStoreVersion path is effectively untested.

**Review by Swatinem — APPROVED** [review]

**Comment by flub** [comment]
@Swatinem so PTAL once more 😄 
I did some manual API requests on an app which has published builds as well to make sure our logic works there as well.  Updated the comments and made a minor adjustment that i expect to be really no-op in all cases, but you never know.

**Review by Swatinem — APPROVED** [review]

**Pull request merged**